L73-java-binderchannel/wireformat.md

```diff
@@ -229,7 +253,53 @@ stream transactions until we receive an acknowledgement message from the transpo
 peer. Each transport sends an acknowledgement for each 16KB of stream transaction
 received, which should avoid blocking the transport most of the time.
 
+## Stream Flow Control
+
+While transport flow control limits consumption of scarce buffer resources at
+the Android/binder layer, stream flow control limits the amount of buffering at
+the gRPC layer. The resulting back pressure lets an application adjust its rate
+of message production to match the remote reader's rate of stream consumption.
+
+### Sender Responsibilities
+Each stream sender (both client and server) must keep track of the amount of
+space remaining in its peer's receive window, decrementing this counter by the
+`bytes data` or `parcelable` size of each `rpc transaction` it sends and
+incrementing it upon receipt of a `stream receive window update`. A stream
```

Comment:
It's worth nothing that parcelables throw a bit of a spanner in the works here, since we can't easily predict how large they'll be.

I'm not too concerned with that though since we never use parcelable messages internally, and if someone uses them they just need to set a super lenient window.

Reply:
This is a great point. I added a new "How to handle Parcelable Message Streams?" section to the design doc for discussion. Quick question though: I know we don't use parcelable messages internally but I noticed that ParcelableInputStream isn't public and we don't expose a Marshaller<T> that can use it. So ... could anybody be using Parcelable message Streams?